cpp/src/structs/utilities.cpp

```diff
@@ -28,16 +26,10 @@
 #include <cudf/utilities/error.hpp>
 #include <cudf/utilities/span.hpp>
 #include <cudf/utilities/traits.hpp>
-#include <cudf/utilities/type_dispatcher.hpp>
-
-#include <rmm/device_buffer.hpp>
 
 #include <thrust/iterator/counting_iterator.h>
 #include <thrust/iterator/transform_iterator.h>
 
-#include <bitset>
-#include <iterator>
-
 namespace cudf::structs::detail {
```

Comment:
This comment is not in the scope of this PR.
Since these functions apply to list, string as well, consider if moving this out of namespace `cudf::structs::` makes sense.